apps/src/Sounds.js

```diff
@@ -235,6 +240,42 @@ Sounds.prototype.checkDidSourcePlay_ = function(source, context, onComplete) {
   );
 };
 
+/**
+ * Registers the bytes of text to speech.
+ * @param {string} language the language the text is spoken in.
+ * @param {string} text the text that is being spoken.
+ * @param {Array.<string>} profaneWords an array of profane words in the text if there are any. Otherwise null.
+ * @param {string} gender the gender of the voice used to speak the text. Optional
+ * @param {Array.<bytes>} bytes the bytes of the speech.
+ */
+Sounds.prototype.registerTextBytes = function(
+  language,
+  text,
+  profaneWords,
+  gender,
+  bytes
+) {
+  this.textBytesByLanguage[language] = this.textBytesByLanguage[language] || {};
+  this.textBytesByLanguage[language][text] =
+    this.textBytesByLanguage[language][text] || {};
+  this.textBytesByLanguage[language][text].profaneWords = profaneWords;
+  if (gender) {
+    this.textBytesByLanguage[language][text][gender] = bytes;
+  }
+};
+
+/**
+ * Retrieves the bytes of text to speech if it exists.
+ * @param {string} language the language the text is spoken in.
+ * @param {string} text the text that is being spoken.
+ * @returns {Object} if text bytes found, null otherwise.
+ */
+Sounds.prototype.getTextBytes = function(language, text) {
+  return (
+    this.textBytesByLanguage.language && this.textBytesByLanguage.language.text
```

Comment:
should this be `[language]` and `[text]`? sorry for messing up dot vs bracket notation before!

Reply:
oops you're right, good catch!